### PR TITLE
#166708134 User should be able to send a Private Message

### PR DESCRIPTION
**Pull request by chizzydavid**
### What does this PR do?

* Enables users send a private message to another user.

### Description of Task to be completed?

* Create PrivateMessage Model, adding all appropriate fields using Sequelize.
* Created a **POST** route to *api/privateMessages*
* Added Validation to this route to check for wrong input.
* Created a *sendMessage* Controller function to handle incoming requests and send the valid user data to the database.
* Wrote unit tests for the *sendMessage* controller method.
* Added Sequelize tests for the PrivateMessage Model.

### Any background context you want to provide?

* User would have to provide a token in order to access this route.

### What are the relevant Github issues?

N/A

### Questions:

N/A

#### PT STORY [#166708134](https://www.pivotaltracker.com/n/projects/2314472/stories/166708134)

### CODE REVIEW

File: server/controller/privateMessage.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import models from '../models';
+
+const { User } = models;
+/**
+ * Handles private message operations
+* */
+class PrivateMessageController {
```
Comment (fortunee) [inline server/controller/privateMessage.js]: Just `MessageController` is fine

File: server/migrations/20190617195952-create-private-message.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    receiverId: {
+      type: Sequelize.INTEGER,
+      allowNull: false,
+      onDelete: 'CASCADE',
+      references: {
+        model: 'Users',
+        key: 'id'
+      }
+    },
+    parentMessageId: {
```
Comment (fortunee) [inline server/migrations/20190617195952-create-private-message.js]: Just out of curiosity here - Why do we need the `parentMessageId` really?

File: server/migrations/20190617195952-create-private-message.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      type: Sequelize.INTEGER,
+      allowNull: false
+    },
+    status: {
+      type: Sequelize.STRING,
+      allowNull: false
+    },
+    createdAt: {
```
Comment (fortunee) [inline server/migrations/20190617195952-create-private-message.js]: Let's have the `updatedAt` field here as well

File: server/models/PrivateMessage.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+module.exports = (sequelize, DataTypes) => {
+  const PrivateMessage = sequelize.define('PrivateMessage', {
```
Comment (fortunee) [inline server/models/PrivateMessage.js]: Just `Message` and not `PrivateMessage`

File: server/middleware/validations/privateMessage.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const { check, validationResult } = require('express-validator/check');
+
+const privateMsgValidator = [
```
Comment (fortunee) [inline server/middleware/validations/privateMessage.js]: Just `MessageValidator` here is fine

**Review by fortunee — CHANGES_REQUESTED** [review]
Please rename the `privateMessage.js` file to just `message.js`

File: server/migrations/20190617195952-create-private-message.js (hunk 2 above)
Comment (chizzydavid) [reply server/migrations/20190617195952-create-private-message.js]: Okay, I was thinking in a situation where a particular message is sent as a reply to another message. 
`parentMessageId` would be the Id of the message that is being replied to.

File: server/migrations/20190617195952-create-private-message.js (hunk 3 above)
Comment (chizzydavid) [reply server/migrations/20190617195952-create-private-message.js]: Okay sure

**Review by fortunee — APPROVED** [review]
LGTM

**Pull request merged**